src/app/fetcher/fullDomFetcher.js

```diff
@@ -42,9 +38,24 @@ export default async function fetch(url, cssSelectors, { retry } = { retry: 0 })
       throw new FetchDocumentError(`Received HTTP code ${statusCode} when trying to fetch '${url}'`);
     }
 
-    await Promise.all(selectors.map(selector => page.waitForSelector(selector, { timeout: config.get('fetcher.waitForElementsTimeout') })));
+    const mimeType = (response.headers() || [])['content-type'];
+
+    if (mimeType.startsWith('text/')) {
+      content = await response.text();
```

Comment:
I reproduced the behavior that was existent on the htmlOnlyFetcher.

`page.content` will return a blank html page even if content is empty
`response.text` returns the exact data retrieved

Reply:
I'm not sure to understand, TBD.